### PR TITLE
bash-completions/rc-service: don't use grep

### PR DESCRIPTION
**Pull request by N-R-K**
Closes: https://github.com/OpenRC/openrc/issues/344

### CODE REVIEW

**Comment by vapier** [comment]
we use `grep` in 2 places, and it looks like we don't need it at all.
```shell
	words="$(rc-service --list | grep ^${cur})"		  # complete for init scripts
	COMPREPLY=($(for i in ${words} ; do \
	[[ ${i} == ${cur}* ]] && echo ${i} ; \
	done))
```

grep is used to filter the output of `rc-service --list`, but then we also manually filter with bash.  it's actually kind of worse in that we first treat the user input as a regex, and then we treat it as a bash glob.  it should be a string in both cases.

```diff
-	words="$(rc-service --list | grep ^${cur})"		  # complete for init scripts
-	COMPREPLY=($(for i in ${words} ; do \
+	COMPREPLY=($(for i in $(rc-service --list) ; do \
-	[[ ${i} == ${cur}* ]] && echo ${i} ; \
+	[[ ${i} == "${cur}"* ]] && echo "${i}" ; \
	done))
```

then we don't need to muck with aliases at all.   i'm not saying we want to respect them, it's just easier to not mess with traps in the first place.

**Comment by N-R-K** [comment]
That seems reasonable. Did some light testing with that patch and it seems to work OK.

File: bash-completion/rc-service [hunk 1]
```diff
@@ -105,8 +105,7 @@ _rc_service()
 	fi
 	if [[ ${action} == '--exists' ]] || [[ ${action} == '-e' ]] || \
 		[[ ${action} == '--resolve' ]]  || [[ ${action} == '-r' ]]; then
-		words="$(rc-service --list | grep ^${cur})"
-		COMPREPLY=($(for i in ${words} ; do \
+		COMPREPLY=($(for i in $(rc-service --list) ; do \
 			[[ ${i} == ${cur}* ]] && echo ${i} ; \
```
Comment (vapier) [inline bash-completion/rc-service:109]: `"${cur}"` needs quoting here too

**Comment by vapier** [comment]
thx bud

**Pull request merged**